LukeMason33/Rotten-Tomatillos
# Feature search filter polish

### Is this a fix or a feature?

Feature.

### What is the change/fix?

This PR adds search bar clearing functionality, and disables genre filtering when the search bar is in use.

### Where should the reviewer start?

Familiarize yourself with the changes made to the search bar value in `search-component.js`, and the `clearInput` function in `app.js`. Additionally, look at where `clearInput` is being applied in the `movie-card-component` button.

### How should this be tested?

In browser, search for a movie in the search bar, click the details button on the movie and visit its page. Once on the single movie page, return to the main page and ensure that the search bar has been cleared, all movies are displaying, and the genre filter has been reset to 'All'. Additionally, once back on the home page, filter movies by genre. Once movies are being filtered by genre, search for a movie in the search bar, and ensure that the genre filter is disabled and is now displaying 'All' genres.
